example/index.js

```diff
@@ -0,0 +1,127 @@
+import { createLightNode } from "js-waku/lib/create_waku";
+import { waitForRemotePeer } from "js-waku/lib/wait_for_remote_peer";
+import * as noise from "@waku/js-noise";
+import QRCode from "qrcode";
+// TODO: Get rid of these
+import hexToArrayBuffer from "hex-to-array-buffer";
+import arrayBufferToHex from "array-buffer-to-hex";
+
+function getPairingInfo() {
+  const urlParts = window.location.href.split("?");
+  if (urlParts.length < 2) return undefined;
+
+  const pairingParts = decodeURIComponent(urlParts[1]).split(":");
+  if (pairingParts.length < 6) throw new Error("invalid pairing information format");
+
+  const qrMessageNameTag = new Uint8Array(hexToArrayBuffer(pairingParts.shift()));
+
+  return new noise.InitiatorParameters(pairingParts.join(":"), qrMessageNameTag);
+}
+
+async function confirmAuthCodeFlow(pairingObj) {
+  const authCode = await pairingObj.getAuthCode();
+  pairingObj.validateAuthCode(confirm("Confirm that authcode is: " + authCode));
+}
+
+async function main() {
+  // Starting the node
+  const node = await createLightNode();
+  await node.start();
+
+  // Dialing a node and wait until it's available
+  const ma =
+    "/dns4/node-01.ac-cn-hongkong-c.wakuv2.test.statusim.net/tcp/443/wss/p2p/16Uiu2HAkvWiyFsgRhuJEb9JfjYxEkoHLgnUQmr1N5mKWnYjxYRVm";
+  await node.dial(ma, ["filter", "lightpush"]);
+  await waitForRemotePeer(node, ["filter", "lightpush"]);
+
+  const sender = {
+    async publish(encoder, msg) {
+      await node.lightPush.push(encoder, msg);
+    },
+  };
+
+  const msgQueue = new Array();
+  const receiver = {
+    async subscribe(decoder) {
+      await node.filter.subscribe([decoder], (wakuMessage) => {
+        msgQueue.push(wakuMessage);
+        // TODO: remove subscription once handshake ends?
+      });
+    },
+    async nextMessage(contentTopic) {
+      if (msgQueue.length != 0) {
+        const oldestMsg = msgQueue.shift();
+        if (oldestMsg.contentTopic === contentTopic) {
+          return oldestMsg;
+        }
+      }
+
+      return new Promise((resolve) => {
+        const interval = setInterval(() => {
+          if (msgQueue.length != 0) {
+            clearInterval(interval);
+            const oldestMsg = msgQueue.shift();
+            if (oldestMsg.contentTopic === contentTopic) {
+              resolve(oldestMsg);
+            }
+          }
+        }, 100);
+      });
+    },
+  };
+
+  const myStaticKey = noise.generateX25519KeyPair();
```

Comment:
This starts to be out of my knowledge.

What is this key? Would we want to save it, backed it up?

Maybe the questions is: waht information needs to be backed up to be able to decrypt old messages?

Reply:
Each user should know its own static key and could be generated with `crypto.generateKeyPairFromSeed`. In the example I'm using a random key. Backing up ephemeral keys (which are an optional parameter when instantiating the pairing object), will need to be done once we implement forward secrecy.

The information that should be backed up can be obtained from the pairing object via the `getHandshakeResult` function, and then to decrypt old messages `WakuPairing.getSecureCodec` can be used. I updated the example with this piece of code:

```js
// The information needs to be backed up to decrypt messages sent with
// codecs generated with the handshake
const contentTopic = pairingObj.contentTopic;
const handshakeResult = pairingObj.getHandshakeResult();

// To restore the codecs:
const codecs = WakuPairing.getSecureCodec(contentTopic, handshakeResult);
```